usb/esp_tinyusb/tinyusb_net.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+
+#include "tinyusb_net.h"
+#include "descriptors_control.h"
+#include "usb_descriptors.h"
+
+#define MAC_ADDR_LEN 6 // In bytes
+
+static SemaphoreHandle_t  Net_Semphore;
+static net_recv_handler_t net_recv_callback;
+
+bool tud_network_wait_xmit(uint32_t ms)
+{
+    if (xSemaphoreTake(Net_Semphore, ms/portTICK_PERIOD_MS) == pdTRUE) {
+        xSemaphoreGive(Net_Semphore);
+        return true;
+    }
+    return false;
+}
+
+void tusb_net_init(void)
+{
+}
+
+esp_err_t tinyusb_net_send(void *buffer, uint16_t len)
+{
+    if (!tud_ready()) {
+        return ESP_FAIL;
+    }
+    
+    if (tud_network_wait_xmit(100)) {
+        /* if the network driver can accept another packet, we make it happen */
+        if (tud_network_can_xmit(len)) {
+            tud_network_xmit(buffer, len);
+        }
+    }
+    return ESP_OK;
+}
+
+esp_err_t tinyusb_net_init(tinyusb_usbdev_t usb_dev, const tinyusb_net_config_t *cfg)
+{
+    (void) usb_dev;
+    uint8_t mac_id;
+    //TODO: Init the network driver
+    vSemaphoreCreateBinary(Net_Semphore);
```

Comment:
Question: Is this binary semaphore used for exclusive access? Would a mutex be more suitable here?